substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/AnalysisObjectScanner.java

```diff
@@ -36,8 +36,15 @@
 
 public class AnalysisObjectScanner extends ObjectScanner {
 
-    public AnalysisObjectScanner(BigBang bigbang, CompletionExecutor executor, ReusableSet scannedObjects) {
-        super(bigbang, executor, scannedObjects);
+    /**
+     * As opposed to the other subclasses of Object scanner, this one is very points-to specific,
+     * therefore it is keeping a reference to BigBang.
+     */
+    private final PointsToAnalysis analysis;
+
+    public AnalysisObjectScanner(PointsToAnalysis bb, CompletionExecutor executor, ReusableSet scannedObjects) {
+        super(bb, executor, scannedObjects);
+        this.analysis = bb;
```

Comment:
This class already has access to `bb` from its super class. It just needs to be cast to `PointsToAnalysis` on some accesses; that could be wrapped in a method to keep the code compact.

Reply:
Good point, I've refactored it.